gRPC-C++.podspec

```diff
@@ -1486,6 +1486,7 @@ Pod::Spec.new do |s|
   end
 
   s.prepare_command = <<-END_OF_COMMAND
+    if [[ "$(uname)" == "Darwin" ]]; then
```

Comment:
I don't understand why this is necessary and why it's the correct solution. Mind explaining in a bit more detail?

Reply:
Firebase uses Cloud's [repository gardener](https://github.com/GoogleCloudPlatform/repository-gardener), which runs jobs on Linux kokoro to update dependencies on open-source repositories. The [script](https://github.com/GoogleCloudPlatform/repository-gardener/blob/master/use-latest-deps-ios.sh) we run uses CocoaPods to actually generate the dependency updates via `pod update`, which invokes `pod install`, which currently fails for gRPC-C++ due to differences in Linux and macOS `sed` and `base64`.

This PR skips the script portion of gRPC-CC++'s `pod install` on Linux systems to allow the installation to succeed.

Reply:
we should probably avoid branching on specific platforms, and instead find out what's causing the sed errors and use a common denominator syntax that would work for both (Darwin/BSD sed & Linux/GNU sed) if possible. See some comparison [here](https://riptutorial.com/sed/topic/9436/bsd-macos-sed-vs--gnu-sed-vs--the-posix-sed-specification)